docs/source/perf/spmd_basic.md

```diff
@@ -61,56 +65,133 @@ SPMD with other distributed libraries.
 
 ### Mesh
 
-For a given cluster of devices, a physical mesh is a representation of
-the interconnect topology.
+The SPMD programming model is built around the concept of a device mesh, commonly
+referred to as a mesh. A device mesh is a logical N-dimensional arrangement of
+compute devices (e.g. TPU cores) where the axes are used as shorthands to
```

Comment:
What does "the axes are used as shorthands to define sharded computation" mean?

Reply:
Expanded to "the axes are used as shorthands to define the devices that participate in a sharded computation".

Basically each axis of the mesh will let you select a bunch of devices. And later when you say I want to shard a tensor over this axis, that means shard (distribute pieces of) the tensor over those devices. So the axis is a shorthand.

Wondering if you had clearer ideas?

Reply:
Basically, rather than saying "do the all-gather for device 2,4,6,8,10,12.......", we can say "do the all-gather over the fsdp axis"